skyportal/handlers/api/photometry.py

```diff
@@ -53,86 +60,130 @@ def post(self):
 
         data = self.get_json()
 
-        if 'mjd' not in data:
-            return self.error('mjd is a required parameter.')
-        if not isinstance(data['flux'], (list, tuple)):
-            for colname in PHOTOMETRY_COLUMNS:
-                data[colname] = [data[colname]]
+        if not isinstance(data, dict):
+            return self.error('Top level JSON must be an instance of `dict`, got '
+                              f'{type(data)}.')
+
+        if allscalar(data):
+            data = [data]
 
         try:
-            lc = pd.DataFrame({colname: data[colname] for colname in PHOTOMETRY_COLUMNS})
+            df = pd.DataFrame(data)
         except ValueError as e:
-            return self.error(f'Improperly formatted input data: {e}')
+            return self.error('Unable to coerce passed JSON to a series of packets. '
+                              f'Error was: "{e}"')
+
+        # pop out thumbnails and process what's left using schemas
```

Comment:
Are thumbnails being handled here anymore, is that functionality removed?

Reply:
I removed it for separation of concerns, thought that is better handled in the `ThumbnailHandler` endpoint.

Reply:
That's fine